tmf/org.eclipse.tracecompass.tmf.core/src/org/eclipse/tracecompass/tmf/core/config/ITmfDataProviderSource.java

```diff
@@ -0,0 +1,54 @@
+/*******************************************************************************
+ * Copyright (c) 2024 Ericsson
+ *
+ * All rights reserved. This program and the accompanying materials are
+ * made available under the terms of the Eclipse Public License 2.0 which
+ * accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *******************************************************************************/
+
+package org.eclipse.tracecompass.tmf.core.config;
+
+import java.util.List;
+
+import org.eclipse.tracecompass.tmf.core.dataprovider.DataProviderManager;
+import org.eclipse.tracecompass.tmf.core.dataprovider.IDataProviderDescriptor;
+import org.eclipse.tracecompass.tmf.core.exceptions.TmfConfigurationException;
+import org.eclipse.tracecompass.tmf.core.trace.ITmfTrace;
+
+/**
+ * Interface for creating data providers. Implementers only return the data provider descriptors.
+ * Instantiating the actual data provider(s) should be done using {@link DataProviderManager}.
+ * @since 9.5
+ */
+public interface ITmfDataProviderSource extends ITmfConfigurationTypesProvider {
+    /**
+     * Create a list of data provider descriptors based on input parameters.
+     *
+     * @param typeId
+     *            The Configuration TypeId
+     * @param trace
+     *            The trace (or experiment) instance
+     * @param jsonParameters
+     *            The configuration parameters as valid json string
+     * @return List of data provider descriptor
+     * @throws TmfConfigurationException
+     *             if an error occurs
+     */
+//    List<IDataProviderDescriptor> createDataProviderDescriptors(String typeId, ITmfTrace trace, Map<String, Object> parameters) throws TmfConfigurationException;
+    List<IDataProviderDescriptor> createDataProviderDescriptors(String typeId, ITmfTrace trace, String jsonParameters) throws TmfConfigurationException;
```

Comment:
The jsonParameter contains the json as string that was specified in configuration source type. I requires a name field and an optional description field. The name will be shown to the user as part of the data provider descriptor it creates.

Notes:

That this interface means that the jsonParameter are assigned to all traces in the experiment. It's not required at the moment to have parameters per trace.

The data providers and descriptors need an ITmfTrace instance in TC.core, so we need to pass the instance. Ideally for configurations, it shouldn't be necessary to have a trace object (that also executes analysis). 

To avoid that there would need to be a separate model representing traces/experiments/analysis like it is done with classic Trace Compass UI model. Refactoring for that would be big undtertaking.